pkg/kubernetes/resources.go

```diff
@@ -45,7 +46,33 @@ func (k *Kubernetes) ResourcesList(ctx context.Context, gvk *schema.GroupVersion
 	return k.manager.dynamicClient.Resource(*gvr).Namespace(namespace).List(ctx, options.ListOptions)
 }
 
+// RESTConfigForGVK returns a copy of the REST config for creating custom clients.
+// Access control is enforced by validating the provided GVK.
+// This is useful for accessing subresources that aren't supported by the dynamic client.
+// The caller is responsible for configuring GroupVersion, APIPath, and NegotiatedSerializer
+// as needed for their specific use case.
+func (k *Kubernetes) RESTConfigForGVK(gvk *schema.GroupVersionKind) (*rest.Config, error) {
+	if k.manager == nil {
+		return nil, fmt.Errorf("kubernetes manager not initialized")
+	}
+
+	// Validate access control for the resource kind
+	if gvk != nil {
+		_, err := k.resourceFor(gvk)
+		if err != nil {
+			return nil, err // Access denied or resource not found
+		}
+	}
+
+	// Return a copy of the REST config to avoid modifying the original
+	return rest.CopyConfig(k.manager.cfg), nil
+}
+
```

Comment:
Hey @lyarwood - sorry this is coming after you've put so much work into this PR! I've been trying to work on how to better handle providing generic access to the API server while still enforcing the MCP server's access control rules. What I'm currently thinking is https://github.com/containers/kubernetes-mcp-server/pull/473, which will provide you with an access controlled rest client that you can use for whatever you want.

Do you think that would be able to replace your custom `RestConfigForGVK` and rest client usage?

cc @manusa - not sure if you want to wait and see if #473 lands, or if you are okay with merging this method for now and refactoring this toolset after #473